Implement GitHub app installation management

bduffany opened the pull request:
- Shows all installations that your linked GitHub account has access to, as well as all installations linked to your org (even if your linked GitHub account doesn't have access to the installation)
  - If not yet linked to BB, show a "Link" button that calls LinkGitHubAppInstallation
  - If linked to BB (any org, even ones you aren't a member of), show an unlink button
    - If not linked to the currently selected org, show a warning

In a separate PR, I might add convenience links for managing each individual installation (via GitHub), but for now that can be done via the "Setup" button then selecting the installation (it's just an extra click)

https://user-images.githubusercontent.com/2414826/229212245-386b24e6-261d-4e99-bf47-89ad632c1272.mp4



---

**Version bump**: Patch <!-- Required. Choose from: Major, Minor, Patch, None -->

<!-- See https://semver.org/#semantic-versioning-specification-semver. Summary:
* Major: Breaking change that causes existing functionality to not work as expected.
* Minor: Non-breaking change that adds functionality (examples: new feature; new API options)
* Patch: Non-breaking change that fixes an issue, improves performance, or refactors
         code.
* None:  Changed files are not included in releases (tests, docs, development setup,
         production configs)
-->

<!-- Optional:
**Related issues**: Fixes #1, Unblocks #2 ...
-->

siggisim commented:
Also, what are your thoughts on hiding the old link UI and warning message unless you already have an old account linked? 

That way it can only be used for unlinking (and not for linking new accounts).

bduffany commented:
> what are your thoughts on only showing repos that are linked to you current org instead? 

I implemented this because I thought people might get confused about why an installation isn't showing up in the org even though they have linked it to BB. But it would simplify things a lot to only show repos linked to the current org, so I've removed this for now and we can always change it later if people think it's confusing.

>  Also, what are your thoughts on hiding the old link UI and warning message unless you already have an old account linked?

I had this same thought earlier today - will do that in a follow-up PR
